src/Service/OpenAI.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+use function base64_encode;
+
+class OpenAI
+{
+    private const PROMPT = <<<'PROMPT'
+    Analyse the picture. It should contain a face or an abstract avatar.
+    Describe the face in detail, ignoring everything aside from the face like a background.
+    Focus on attributes that can help compare the face to other faces and determine whether the faces are similar.
+    If the picture contains multiple faces, describe each face individually.
+    Only return a list of individual attributes that can be used for comparison.
+    PROMPT;
```

Comment:
We can probably tune this prompt a bit, but for now I've gotten pretty good results. For example, here's the description it generated for my GitHub Avatar:
> - Face shape: Oval
> - Hair color: Red
> - Hair style: Short, combed back
> - Facial hair: Full beard and mustache, mixed color (red and gray)
> - Skin tone: Light
> - Eye color: Blue
> - Eyewear: Large, black-rimmed glasses with a unique angular shape
> - Forehead: Smooth with slight wrinkles near the eyebrows
> - Eyebrows: Medium thickness, light brown
> - Nose: Medium size, straight
> - Lips: Medium fullness, smiling
> - Teeth: Visible, white and well-aligned

We may potentially want more structured results in order to improve the vector search, but I think this is good for now.